test/jdk/java/net/httpclient/http2/PushPromiseContinuation.java

```diff
@@ -146,6 +146,21 @@ public void testTwoContinuations() {
         verify();
     }
 
+    @Test
+    public void testThreeContinuations() {
+        continuationCount = 3;
+        HttpClient client = HttpClient.newHttpClient();
+
+        // Carry out request
+        HttpRequest hreq = HttpRequest.newBuilder(uri).version(HttpClient.Version.HTTP_2).GET().build();
+        CompletableFuture<HttpResponse<String>> cf =
+                client.sendAsync(hreq, HttpResponse.BodyHandlers.ofString(UTF_8), pph);
+        cf.join();
```

Comment:
I'd suggest checking that we received the expected response code + body too

Reply:
Further changes made to check response codes and bodiesof both the response and push promise.